f3/granite.go

```diff
@@ -782,7 +791,24 @@ func (i *instance) broadcast(round uint64, step Phase, value ECChain, ticket Tic
 // The delay duration increases with each round.
 // Returns the absolute time at which the alarm will fire.
 func (i *instance) alarmAfterSynchrony() float64 {
-	timeout := i.host.Time() + i.config.Delta + (float64(i.round) * i.config.DeltaRate)
+	delta := i.config.Delta
+
+	if i.round >= 2 {
+		delta += i.config.DeltaExtra
+	}
+
+	if i.round >= 5 {
+		delta *= i.deltaRate
+		i.deltaRate *= i.config.DeltaRate
```

Comment:
Please remove the mutable state variable, you can just count the rounds. 

`delta = delta * deltaRate ^ (i.round - 4)` or similar.

Reply:
Done.